SmartDeviceLink/SDLLockScreenPresenter.m

```diff
@@ -8,48 +8,101 @@
 
 #import "SDLLockScreenPresenter.h"
 
+#import "SDLLogMacros.h"
+#import "SDLScreenshotViewController.h"
+#import "SDLStreamingMediaManagerConstants.h"
+
 
 NS_ASSUME_NONNULL_BEGIN
 
 @interface SDLLockScreenPresenter ()
 
+@property (strong, nonatomic) SDLScreenshotViewController *screenshotViewController;
+@property (strong, nonatomic) UIWindow *lockWindow;
+
 @end
 
 
 @implementation SDLLockScreenPresenter
 
+- (instancetype)init {
+    self = [super init];
+    if (!self) { return nil; }
+
+    CGRect screenFrame = [[UIScreen mainScreen] bounds];
+    _lockWindow = [[UIWindow alloc] initWithFrame:screenFrame];
+    _screenshotViewController = [[SDLScreenshotViewController alloc] init];
+    _lockWindow.rootViewController = _screenshotViewController;
```

Comment:
Do I understand correctly that the lockscreen is now window-based, instead of trying to find the top-level VC and presenting over it (we were working on a proposal for this change)?

Reply:
That is correct, this was included in the CarWindow proposal.